eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,497 @@
+              inputs:
+                pwsh: true
+                filePath: $(Build.SourcesDirectory)/eng/versioning/verify_release_set.ps1
+                arguments: >
+                  -ServiceDirectory $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                  -ArtifactsList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId, uberJar | Where-Object -Not "uberJar")
+
+        - deployment: TagRepository
+          displayName: "Create release tag"
+          condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
+          environment: github
+          dependsOn: VerifyReleaseVersion
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: windows-2019
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                  - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - download: current
+                      displayName: 'Download Artifact: ${{parameters.ArtifactName}}-signed'
+                      artifact: ${{parameters.ArtifactName}}-signed
+                      patterns: ${{artifact.groupId}}/${{artifact.name}}/**
+                    - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
+                      parameters:
+                        ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
+                        PackageRepository: Maven
+                        ReleaseSha: $(Build.SourceVersion)
+
+        - deployment: PublishPackage
+          displayName: "Publish to Maven Central"
+          condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
+          environment: maven
+          dependsOn: TagRepository
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: windows-2019
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - checkout: azure-sdk-build-tools
+                    path: azure-sdk-build-tools
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                    parameters:
+                      SkipDefaultCheckout: true
+                      Repositories:
+                        - Name: Azure/azure-sdk-for-java
+                          Commitish: $(Build.SourceVersion)
+                          WorkingDirectory: $(Pipeline.Workspace)/azure-sdk-for-java
+                  - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
I know we talked about figuring out a way to run these in parallel but before we over optimize on that we should make the publish script handle multiple artifacts.

Reply:
@weshaggard there's a couple of reasons why I don't want to make any changes to the publish script. Right now, I think we're better off holding off on any changes to the publish script until we know what ESRP looks like and rewriting it at that time.